File: client/src/types/employee.ts
export const EmployeeStatus = {
  ACTIVE: "active",
  INACTIVE: "inactive",
  ON_LEAVE: "on_leave",
} as const;

export type EmployeeStatus = (typeof EmployeeStatus)[keyof typeof EmployeeStatus];

export interface Role {
  id: string;
  name: string;
  defaultHourlyRate: number;
  createdAt: string;
  updatedAt: string;
}

export interface RoleCreate {
  name: string;
  defaultHourlyRate: number;
}

export interface Employee {
  id: string;
  name: string;
  email: string;
  roleId: string;
  roleName: string;
  hourlyRate: number | null;
  effectiveHourlyRate: number;
  jobPercentage: number;
  targetUtilizationPercentage: number;
  status: EmployeeStatus;
  department: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface EmployeeCreate {
  name: string;
  email: string;
  roleId: string;
  hourlyRate?: number;
  jobPercentage: number;
  targetUtilizationPercentage: number;
  department?: string;
}

export interface EmployeeUpdate {
  name?: string;
  email?: string;
  roleId?: string;
  hourlyRate?: number | null;
  jobPercentage?: number;
  targetUtilizationPercentage?: number;
  status?: EmployeeStatus;
  department?: string | null;
}

File: backend/app/schemas/employee.py
from datetime import datetime
from decimal import Decimal
from uuid import UUID

from app.enums.employee_status import EmployeeStatus
from app.schemas.base import JsonModel


class EmployeeCreate(JsonModel):
    name: str
    email: str
    role_id: UUID
    hourly_rate: Decimal | None = None
    job_percentage: int = 100
    target_utilization_percentage: int = 100
    department: str | None = None


class EmployeeUpdate(JsonModel):
    name: str | None = None
    email: str | None = None
    role_id: UUID | None = None
    hourly_rate: Decimal | None = None
    job_percentage: int | None = None
    target_utilization_percentage: int | None = None
    status: EmployeeStatus | None = None
    department: str | None = None


class EmployeeResponse(JsonModel):
    id: UUID
    name: str
    email: str
    role_id: UUID
    role_name: str
    hourly_rate: Decimal | None
    effective_hourly_rate: Decimal
    job_percentage: int
    target_utilization_percentage: int
    status: EmployeeStatus
    department: str | None
    created_at: datetime
    updated_at: datetime
